Source/Core/Core/Slippi/SlippiMatchmaking.cpp

```diff
@@ -301,58 +327,58 @@ void SlippiMatchmaking::startMatchmaking()
 	    return;
 	}*/
 
-	// The following code attempts to fetch the LAN IP such that when remote IPs match, the
-	// LAN IP can be tried in order to establish a connection in the case where the players
-	// don't have NAT loopback which allows that type of connection.
-	// Right now though, the logic would replace the WAN IP with the LAN IP and if the LAN
-	// IP connection didn't work but WAN would have, the players can no longer connect.
-	// Two things need to happen to improtve this logic:
-	// 1. The connection must be changed to try both the LAN and WAN IPs in the case of
-	//    matching WAN IPs
-	// 2. The process for fetching LAN IP must be improved. For me, the current method
-	//    would always fetch my VirtualBox IP, which is not correct. I also think perhaps
-	//    it didn't work on Linux/Mac but I haven't tested it.
-	// I left this logic on for now under the assumption that it will help more people than
-	// it will hurt
+	// Determine local IP address. We can attempt to connect to our opponent via
+	// local IP address if we have the same external IP address. The following
+	// scenarios can cause us to have the same external IP address:
+	// - we are connected to the same LAN
+	// - we are connected to the same VPN node
+	// - we are behind the same CGNAT
 	char lanAddr[30] = "";
-
-	char host[256];
-	char *IP;
-	struct hostent *host_entry;
-	int hostname;
-	hostname = gethostname(host, sizeof(host)); // find the host name
-	if (hostname == -1)
+	if (SConfig::GetInstance().m_slippiForceLanIp)
 	{
-		ERROR_LOG(SLIPPI_ONLINE, "[Matchmaking] Error finding LAN address");
+		WARN_LOG(SLIPPI_ONLINE, "[Matchmaking] Overwriting LAN IP sent with configured address");
+		sprintf(lanAddr, "%s:%d", SConfig::GetInstance().m_slippiLanIp.c_str(), m_hostPort);
 	}
 	else
 	{
-		host_entry = gethostbyname(host); // find host information
-		if (host_entry == NULL || host_entry->h_addrtype != AF_INET)
-		{
-			ERROR_LOG(SLIPPI_ONLINE, "[Matchmaking] Error finding LAN host");
-		}
+		enet_uint32 localAddress = getLocalAddress(&addr);
+		if (localAddress != 0)
+			sprintf(lanAddr, "%s:%d", inet_ntoa(*(struct in_addr *)&localAddress), m_hostPort);
 		else
 		{
-			// Fetch the last IP (because that was correct for me, not sure if it will be for all)
-			int i = 0;
-			while (host_entry->h_addr_list[i] != 0)
+			// Fallback: just try the last local IP address listed. They seem to be
```

Comment:
I feel like all of the local address stuff might as well be moved to its own function?

Reply:
I tried to leave the old determination method alone to have a nicer diff but it didn't work out I guess. You want me to move just this old block?

Reply:
cleaned up the fallback method a bit and factored it out to a separate method. good call looks a lot nicer now imo